up-streamer/src/ustreamer.rs

```diff
@@ -420,22 +466,65 @@ impl UStreamer {
     /// * name - Used to uniquely identify this UStreamer in logs
     /// * message_queue_size - Determines size of channel used to communicate between `ForwardingListener`
     ///                        and the worker tasks for each currently endpointd `UTransport`
-    pub fn new(name: &str, message_queue_size: u16) -> Self {
+    /// * usubscription - Subscription service which will be used to store subscription info for topics.
+    pub fn new(name: &str, message_queue_size: u16, usubscription: Arc<dyn USubscription>) -> Self {
         let name = format!("{USTREAMER_TAG}:{name}:");
         // Try to initiate logging.
         // Required in case of dynamic lib, otherwise no logs.
         // But cannot be done twice in case of static link.
+        std::env::set_var("RUST_LOG", "debug");
         let _ = env_logger::try_init();
         debug!(
             "{}:{}:{} UStreamer created",
             &name, USTREAMER_TAG, USTREAMER_FN_NEW_TAG
         );
 
+        let uuri: UUri = UUri {
+            authority_name: "*".to_string(),
+            ue_id: 0x0000_FFFF,     // any instance, any service
+            ue_version_major: 0xFF, // any
+            resource_id: 0xFFFF,    // any
+            ..Default::default()
+        };
+
+        let details_vector = Vec::new();
+        let subscriber_info = SubscriberInfo {
+            uri: Some(uuri.clone()).into(),
+            details: details_vector,
+            ..Default::default()
+        };
+
+        let notifications_register_request = NotificationsRequest {
+            topic: Some(uuri).into(),
+            ..Default::default()
+        };
+        let _ = task::block_on(
```

Comment:
This isn't really needed, since this would look totally different.

Maybe put a:
```rust
// TODO: Create a NotificationsRequest and send over host transport
```

Reply:
done